x/pairing/keeper/epoch_payment.go

```diff
@@ -0,0 +1,81 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/utils"
+	"github.com/lavanet/lava/x/pairing/types"
+)
+
+// AddEpochPayment adds a new epoch payment and returns the updated CU used between provider and project
+func (k EpochCuCache) AddEpochPayment(ctx sdk.Context, chainID string, epoch uint64, project string, provider string, cu uint64, sessionID uint64) uint64 {
+	// register new epoch session (not checking double spend because it's already checked before calling this function)
+	k.SetUniqueEpochSession(ctx, epoch, provider, project, chainID, sessionID)
+
+	// update provider serviced CU
+	pec, found := k.GetProviderEpochCuCached(ctx, epoch, provider, chainID)
+	if !found {
+		pec = types.ProviderEpochCu{ServicedCu: cu}
+	} else {
+		pec.ServicedCu += cu
+	}
+	k.SetProviderEpochCuCached(ctx, epoch, provider, chainID, pec)
+
+	// update provider CU for the specific project
+	pcec, found := k.GetProviderConsumerEpochCuCached(ctx, epoch, provider, project, chainID)
+	if !found {
+		pcec = types.ProviderConsumerEpochCu{Cu: cu}
+	} else {
+		pcec.Cu += cu
+	}
+	k.SetProviderConsumerEpochCuCached(ctx, epoch, provider, project, chainID, pcec)
+	return pcec.Cu
+}
+
+// Function to remove epoch payment objects from deleted epochs (older than the chain's memory)
+func (k Keeper) RemoveOldEpochPayments(ctx sdk.Context) {
+	epochsToDelete := k.epochStorageKeeper.GetDeletedEpochs(ctx)
+	for _, epoch := range epochsToDelete {
+		k.RemoveAllEpochPaymentsForBlockAppendAdjustments(ctx, epoch)
+	}
+}
+
+// Function to remove all epoch payments objects from a specific epoch
+func (k Keeper) RemoveAllEpochPaymentsForBlockAppendAdjustments(ctx sdk.Context, epochToDelete uint64) {
+	// remove all unique epoch sessions
+	k.RemoveAllUniqueEpochSession(ctx, epochToDelete)
+
+	// remove all provider epoch cu
+	k.RemoveAllProviderEpochCu(ctx, epochToDelete)
+
+	keys, pcecs := k.GetAllProviderConsumerEpochCu(ctx, epochToDelete)
```

Comment:
can this be 1 list instead of 2?

Reply:
done. see https://github.com/lavanet/lava/commit/a04a6b5b8cfc92c1107d57532077e7c0ed9defda